src/Commands/FailedJobReasons.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace ActiveCollab\JobQueue\Commands;
+
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Exception;
+
+/**
+ * @package ActiveCollab\JobQueue\Commands
+ */
+class FailedJobReasons extends Command
+{
+    /**
+     * Configure command
+     */
+    protected function configure()
+    {
+        parent::configure();
+
+        $this->setName('failed_job_reasons')
+             ->addArgument('type', InputArgument::REQUIRED, 'Name of the job type, or part that matches only one job')
+             ->setDescription('List distinct reasons why a particular job type failed');
+    }
+
+    /**
+     * @param  InputInterface  $input
+     * @param  OutputInterface $output
+     * @return int
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        try {
+            $connection = $this->getDatabaseConnection($input);
+
+            $event_type_names = $connection->executeFirstColumn('SELECT DISTINCT(`type`) FROM `jobs_queue_failed` WHERE `type` LIKE ?', '%' . $input->getArgument('type') . '%');
```

Comment:
Consider extracting to Queue interface, instead of directly querying the database